prover/src/trace/poly_table.rs

```diff
@@ -98,13 +100,16 @@ impl<E: FieldElement> TracePolyTable<E> {
     }
 
     /// Returns an iterator over the polynomials of the main trace segment.
-    pub fn main_trace_polys(&self) -> ColumnIter<E::BaseField> {
+    pub fn main_trace_polys(&self) -> impl Iterator<Item = &[E::BaseField]> {
         self.main_segment_polys.columns()
     }
 
-    /// Returns an iterator over the polynomials of all auxiliary trace segments.
-    pub fn aux_trace_polys(&self) -> MultiColumnIter<E> {
-        MultiColumnIter::new(self.aux_segment_polys.as_slice())
+    /// Returns an iterator over the polynomials of the auxiliary trace segment.
+    pub fn aux_segment_polys(&self) -> impl Iterator<Item = &[E]> {
```

Comment:
Similar to the last comment: `aux_trace_polys` would be more consistent with `main_trace_polys`.

Reply:
fixed